app/models/storage_location.rb

```diff
@@ -77,6 +77,14 @@ def self.items_inventoried(organization, inventory = nil)
     end
   end
 
+  def self.with_transfers_to(organization)
+    joins(:transfers_to).where(organization_id: organization.id).distinct.order(:name)
```

Comment:
You shouldn't need the `where` clause since the transfer has a `from_id` and `to_id` which powers the association.

Reply:
Without the `where` clause, wouldn't we be grabbing every storage location that has at least one transfer from/to that location? Don't we only want the storage locations related to a certain organization?

![Screenshot from 2024-10-28 13-30-41](https://github.com/user-attachments/assets/d953806e-4961-42d4-a503-4248a4af0bd5)

Reply:
Storage locations belong to an organization and so do transfers. So if you have a transfer, by definition only storage locations belonging to the same organization can be involved in it.

Reply:
Sorry I still don't understand why this `where` clause is unnecessary. We are only joining the `transfers` and `storage_locations` tables on the `storage_locations` primary key and the `transfers` foreign keys `from_id` and `to_id`. So it will return records for different organizations unless we specify an `organization_id` somewhere. (Or like if we moved the method to be an instance method on `Organization` so the id is specified by default.)

To illustrate in rails console in development:
1. Organization 1 has a transfer from storage location 1 to storage location 2. Organization 2 has a transfer from storage location 3 to storage location 4.
![Screenshot from 2024-11-13 17-43-45](https://github.com/user-attachments/assets/18c15733-0bfe-4b8b-b1ac-953cfac6b0ac)
2. If I remove the `where` clause, then I get all the storage locations that have been transferred to (locations 2 and 4), which correspond to different organizations.
![Screenshot from 2024-11-13 17-44-12](https://github.com/user-attachments/assets/868e3076-2b80-4375-926c-62e96d4536f2)
3. When I add back in the `where` clause, I see only storage locations that apply to a certain organization which I thought is what we wanted.
![Screenshot from 2024-11-13 17-44-49](https://github.com/user-attachments/assets/0e88237f-4f2a-473f-b97b-460893cc3b6f)

Am I missing something?

Reply:
Nope, I was missing the fact that this was a class-level method instead of an instance-level method.

I think this could be clearer if you made these scopes instead of class-level methods.

Reply:
Ok changed them to be scopes instead :+1: